### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by jamesgk**
This should fix:
https://github.com/googlei18n/fontmake/issues/15
https://github.com/googlei18n/fontmake/issues/14

@adrientetar @brawer whoever would like to take a look, please do.

### CODE REVIEW

**Comment by davelab6** [comment]
@schriftgestalt where is a complete list of all known `customParameter` keys?

**Comment by schriftgestalt** [comment]
There is a file Glyphs.app/Contents/Frameworks/GlyphsCore.framework/Versions/A/Resources/customParameterTypes.plist
and a file that has all allowed keys for font, master and instance (customFontParameters ...).

**Comment by davelab6** [comment]
So unlike FontData.xml you are okay for that list of keys to be libre?

**Comment by adrientetar** [comment]
Lgtm

**Comment by schriftgestalt** [comment]
> So unlike FontData.xml you are okay for that list of keys to be libra?

Yes. That is mostly from The openType and UFO spec. You can probably ignore the ones that start with an uppercase letter as those influence the behavior of Glyphs and are not exported directly. 

there are some keys included in the patch that are from the 'types' file (macintoshFONDFamilyID...) but are not in any of the other three files. That means that the value will not be used by Glyphs and unlikely to be found in a .glyphs file (only if imported from FLS and left there).

> Parse and use time offsets

The time for `lastChange` is always written in GMT (no time zone offset) to avoid diffs/changes by saving in different parts of the world. So the old implementation would be sufficient. Only files from Glyphs 1 might have time zone offsets.

**Comment by davelab6** [comment]
On 16 February 2016 at 18:03, Georg <notifications@github.com> wrote:

> The time for lastChange is always written in GMT (no time zone offset) to
> avoid diffs/changes by saving in different parts of the world. So the old
> implementation would be sufficient. Only files from Glyphs 1 might have
> time zone offsets.
>
That sounds good (to drop it for worldwide collaboration)

File: Lib/glyphs2ufo/casting.py [hunk 1]
```diff
@@ -295,16 +301,37 @@ def feature_syntax(string):
 def custom_params(param_list):
     """Cast some known data in custom parameters."""
 
-    int_params = ('typoAscender', 'typoDescender', 'typoLineGap', 'winAscent',
-                  'winDescent', 'hheaAscender', 'hheaDescender', 'hheaLineGap',
-                  'underlinePosition', 'underlineThickness')
-    intlist_params = ('fsType', 'openTypeOS2Type')
+    int_params = (
```
Comment (davelab6) [inline Lib/glyphs2ufo/casting.py]: Here are the missing int_params:
```
int_params =+ ('CJK Grid Horizontal'
'CJK Grid Vertical'
'CJK Grid'
'Grid Spacing'
'Post Table Type'
'ascender'
'blueShift'
'capHeight'
'descender'
'openTypeHheaAscender'
'openTypeHheaCaretSlopeRise'
'openTypeHheaCaretSlopeRun'
'openTypeHheaDescender'
'openTypeHheaLineGap'
'openTypeOS2TypoAscender'
'openTypeOS2TypoDescender'
'openTypeOS2TypoLineGap'
'openTypeOS2WeightClass'
'openTypeOS2WidthClass'
'openTypeOS2WinAscent'
'openTypeOS2WinDescent'
'openTypeVheaCaretOffset'
'openTypeVheaCaretSlopeRise'
'openTypeVheaCaretSlopeRun'
'openTypeVheaVertTypoAscender'
'openTypeVheaVertTypoDescender'
'openTypeVheaVertTypoLineGap'
'postscriptUnderlinePosition'
'postscriptUnderlineThickness'
'shoulderHeight'
'smallCapHeight'
'unitsPerEm'
'vheaVertAscender'
'vheaVertDescender'
'vheaVertLineGap')
```

File: Lib/glyphs2ufo/casting.py [hunk 2]
```diff
@@ -295,16 +301,37 @@ def feature_syntax(string):
...
+        'openTypeOS2SuperscriptYOffset', 'openTypeOS2SuperscriptYSize',
+        'postscriptBlueFuzz', 'postscriptBlueShift', 'postscriptDefaultWidthX',
+        'postscriptSlantAngle', 'postscriptUniqueID',
+        'postscriptWindowsCharacterSet',
+        'typoAscender', 'typoDescender', 'typoLineGap',
+        'underlinePosition', 'underlineThickness',
+        'winAscent', 'winDescent', 'year')
+    float_params = (
+        'postscriptBlueScale')
+    truthy_params = (
```
Comment (davelab6) [inline Lib/glyphs2ufo/casting.py]: Here is the complete list:
```
truthy_params = ('Add missing symbol glyphs', 'Autohint', 'Compact File Storage', 'Compatible Name Table', 'CompatibleNameTable', 'Disable Subroutines', 'DisableAllAutomaticBehaviour', 'Don’t use Production Names', 'Has WWS Names', 'Keep Overlapping Components', 'Link Metrics With First Master', 'Make morx table', 'Remove post names for webfonts', 'Save as TrueType', 'TrueType Keep GlyphOrder', 'Use Extension Kerning', 'Use Typo Metrics', 'Webfont Only', 'isFixedPitch', 'postscriptForceBold', 'postscriptIsFixedPitch')

File: Lib/glyphs2ufo/casting.py [hunk 3]
```diff
@@ -295,16 +301,37 @@ def feature_syntax(string):
...
+        'postscriptForceBold', 'postscriptIsFixedPitch')
+    intlist_params = (
```
Comment (davelab6) [inline Lib/glyphs2ufo/casting.py]: Here is the complete list: 
```
('Disable Masters', 
'Instance Preview', 
'Keep Glyphs', 
'Reencode Glyphs', 
'Remove Classes', 
'Remove Features', 
'Remove Glyphs', 
'Rename Glyphs', 
'TTFStemSetting', 
'TTFStems', 
'Webfont Formats', 
'codePageRanges', 
'fsType', 
'glyphOrder', 
'openTypeHeadFlags', 
'openTypeOS2CodePageRanges', 
'openTypeOS2FamilyClass', 
'openTypeOS2Panose', 
'openTypeOS2Selection', 
'openTypeOS2Type', 
'openTypeOS2UnicodeRanges', 
'panose', 
'postscriptBlueValues', 
'postscriptFamilyBlues', 
'postscriptFamilyOtherBlues', 
'postscriptOtherBlues', 
'postscriptStemSnapH', 
'postscriptStemSnapV', 
'unicodeRanges')
```

**Comment by jamesgk** [comment]
> > The time for lastChange is always written in GMT (no time zone offset) to
> > avoid diffs/changes by saving in different parts of the world. So the old
> > implementation would be sufficient. Only files from Glyphs 1 might have
> > time zone offsets.

> That sounds good (to drop it for worldwide collaboration)

Well, this new implementation still works fine for timestamps with no offset, and I don't think it hurts to have it for old files. I'd rather just keep it. If a Glyphs file is updated, the offset will be removed anyways.

**Comment by davelab6** [comment]
> I'd rather just keep it

Sure! :8ball:

**Comment by behdad** [comment]
lgtm!

**Pull request merged**